### PR TITLE
[ENG-610] Add draft metadata route

### PR DESCRIPTION
**Pull request by baylee-d**
- Ticket: https://openscience.atlassian.net/browse/ENG-610
- Feature flag: `N/A`

## Purpose

To add a bare-bones metadata route and template to start work on metadata fields.

## Summary of Changes

- Add `drafts.draft.metadata` route
    - Add template, style, controller, route files
- Update registries unit test to check that route exists

## Side Effects

This shouldn't affect any existing code.

## QA Notes

- `N/A` -> This really shouldn't affect any pre-existing code since it's adding a new route/template/controller to the codebase.  It isn't accessible at the moment outside of typing in the direct url.

### CODE REVIEW

File: lib/registries/addon/drafts/draft/metadata/controller.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import RouterService from '@ember/routing/router-service';
+import { inject as service } from '@ember/service';
+
+import DraftRegistration from 'ember-osf-web/models/draft-registration';
+import Media from 'ember-responsive';
+
+import NodeModel from 'ember-osf-web/models/node';
+import { DraftPageRouteModel } from './route';
+
+export default class RegistriesDrat extends Controller {
```
Comment (futa-ikeda) [inline lib/registries/addon/drafts/draft/metadata/controller.ts]: Drat? also, should this be `RegistriesDraftMetadata`?

File: lib/registries/addon/drafts/draft/metadata/controller.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    @service media!: Media;
+    @service router!: RouterService;
+
+    model!: DraftPageRouteModel;
```
Comment (futa-ikeda) [inline lib/registries/addon/drafts/draft/metadata/controller.ts]: Should this be using the page route model?

File: lib/registries/addon/drafts/draft/metadata/route.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import { inject as service } from '@ember/service';
+import { TaskInstance } from 'ember-concurrency';
+import DS from 'ember-data';
+
+import DraftRegistration from 'ember-osf-web/models/draft-registration';
+import Analytics from 'ember-osf-web/services/analytics';
+
+import { DraftRouteModel } from '../route';
+
+export interface DraftPageRouteModel {
```
Comment (futa-ikeda) [inline lib/registries/addon/drafts/draft/metadata/route.ts]: Looks like this is leftover from the page route

File: lib/registries/addon/drafts/draft/metadata/route.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    draftId: string;
+    taskInstance: TaskInstance<DraftRegistration>;
+    pageIndex?: number;
+    page: string;
+}
+
+export default class DraftRegistrationPageRoute extends Route {
```
Comment (futa-ikeda) [inline lib/registries/addon/drafts/draft/metadata/route.ts]: Update the name here please

File: lib/registries/addon/drafts/draft/metadata/styles.scss [hunk 5]
```diff
@@ -0,0 +1,51 @@
+.ContentBackground {
```
Comment (fabmiz) [inline lib/registries/addon/drafts/draft/metadata/styles.scss]: these styles are not necessary as they apply to `OsfLayout` and parent route (draft route) will take care of the layout.

**Comment by adlius** [comment]
Are we keeping the `draft` route under `drafts` route or is it something that will go away?

**Comment by fabmiz** [comment]
I think `registries.drafts` will be used for the users drafts list view further down the line. It sounds right to have the draft detail under it. Anyways, not a blocker here.

**Pull request merged**